kernel/tests/write.rs

```diff
@@ -1813,3 +1814,252 @@ async fn test_ict_commit_e2e() -> Result<(), Box<dyn std::error::Error>> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_remove_files_adds_expected_entries() -> Result<(), Box<dyn std::error::Error>> {
+    // This test verifies that Remove actions generated from scan metadata contain all expected fields
+    // from the Remove struct (defined in kernel/src/actions/mod.rs).
+    //
+    // This test uses the table-with-dv-small dataset which contains files with tags and deletion vectors.
+    //
+    // Not populated in the dataset are (covered by row_tracking tests):
+    // baseRowId (optional i64)
+    // defaultRowCommitVersion (optional i64)
+    use std::path::PathBuf;
+
+    let _ = tracing_subscriber::fmt::try_init();
+
+    let tmp_dir = tempdir()?;
+    let tmp_table_path = tmp_dir.path().join("table-with-dv-small");
+    let source_path = std::fs::canonicalize(PathBuf::from("./tests/data/table-with-dv-small/"))?;
```

Comment:
afaict it's okay to assume we're in the crate root because that's how cargo works, but I get nervous about relative paths like this. I don't have a better suggestion right now though, and it's a test so I think this is okay.

We should really have programatic generation of tables though

Reply:
Yeah, we were missing some part specifically, DV writing (coming in the next PR).  I can try to cleanup some of this after it is merged.